### PR TITLE
Update Advanced CI overview and related pages

### PR DESCRIPTION
**Pull request by nghi-ly**
## What are you changing in this pull request and why?

Reorganized the Advanced CI docs  

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [x] Needs PM review
- [x] Needs Legal review

### CODE REVIEW

**Review by reubenmc — APPROVED** [review]
LGTM

File: website/docs/docs/deploy/about-ci.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+title: "About continuous integration (CI) in dbt Cloud"
+sidebar_label: "About continuous integration"
+pagination_prev: null
+pagination_next: "docs/deploy/continuous-integration"
+hide_table_of_contents: true
+---
+
+Use [CI jobs](/docs/deploy/ci-jobs) in dbt Cloud to set up automation for testing code changes before merging to production. And, [enable Advanced CI features](/docs/dbt-cloud-environments#account-access-to-advanced-ci-features) for these jobs to evaluate whether the code changes are producing the appropriate data changes you want by reviewing the comparison differences dbt provides.
```
Comment (nataliefiann) [inline website/docs/docs/deploy/about-ci.md]:
```suggestion
Use [CI jobs](/docs/deploy/ci-jobs) in dbt Cloud to set up automation for testing code changes before merging to production. Additionally, [enable Advanced CI features](/docs/dbt-cloud-environments#account-access-to-advanced-ci-features) for these jobs to evaluate whether the code changes are producing the appropriate data changes you want by reviewing the comparison differences dbt provides.
```

File: website/docs/docs/deploy/continuous-integration.md [hunk 2]
```diff
@@ -31,7 +31,11 @@ dbt Cloud deletes the temporary schema from your <Term id="data-warehouse" /> w
 
 The [dbt Cloud scheduler](/docs/deploy/job-scheduler) executes CI jobs differently from other deployment jobs in these important ways:
 
-<Expandable alt_header="Concurrent CI checks">
+- **Concurrent CI checks** &mdash; CI runs triggered by the same dbt Cloud CI job execute concurrently (in parallel), when appropriate
```
Comment (nataliefiann) [inline website/docs/docs/deploy/continuous-integration.md]:
```suggestion
- **Concurrent CI checks** &mdash; CI runs triggered by the same dbt Cloud CI job execute concurrently (in parallel), when appropriate.
```

File: website/docs/docs/deploy/continuous-integration.md [hunk 3]
```diff
@@ -31,7 +31,11 @@ dbt Cloud deletes the temporary schema from your <Term id="data-warehouse" /> w
...
+- **Smart cancellation of stale builds** &mdash; Automatically cancels stale, in-flight CI runs when there are new commits to the PR
```
Comment (nataliefiann) [inline website/docs/docs/deploy/continuous-integration.md]:
```suggestion
- **Smart cancellation of stale builds** &mdash; Automatically cancels stale, in-flight CI runs when there are new commits to the PR.
```

File: website/docs/docs/deploy/continuous-integration.md [hunk 4]
```diff
@@ -31,7 +31,11 @@ dbt Cloud deletes the temporary schema from your <Term id="data-warehouse" /> w
...
+- **Run slot treatment** &mdash; CI runs don't consume a run slot
```
Comment (nataliefiann) [inline website/docs/docs/deploy/continuous-integration.md]:
```suggestion
- **Run slot treatment** &mdash; CI runs don't consume a run slot.
```

**Review by nataliefiann — APPROVED** [review]
Hiya @nghi-ly 

This PR is great! Thank you for creating this! I've approved this for you. I've left a few non-blocking suggestions but other than that, I have no pressing recommendations or suggestions.

Kind Regards
Natalie

**Pull request merged**